src/NHibernate/Proxy/Map/MapProxyFactory.cs

```diff
@@ -10,12 +10,22 @@ public class MapProxyFactory : IProxyFactory
 	{
 		private string entityName;
 
-		#region IProxyFactory Members
+		[Obsolete("Please use constructor accepting entityName instead.")]
+		public MapProxyFactory()
+		{
+		}
 
+		public MapProxyFactory(string entityName)
+		{
+			this.entityName = entityName;
+		}
+
+		[Obsolete("Please use constructor accepting entityName instead.")]
 		public void PostInstantiate(string entityName, System.Type persistentClass, ISet<System.Type> interfaces,
 																MethodInfo getIdentifierMethod, MethodInfo setIdentifierMethod,
 																IAbstractComponentType componentIdType)
 		{
+			//6.0 TODO: throw NotSupportedException
```

Comment:
Strange TODO inside obsolete method. Shouldn't we simply drop this method in 6.0? And when obsoleting something we also tend to add `//Since 5.3 comment`

Reply:
It’s not strange. This method would stay on the class (with additional argument) because of the interface, but shall not be supported.

Yeah, I’ve forgot about “since 5.3”

Reply:
That's not so obvious that new override is meant. But why don't add override now?